src/infoview.ts

```diff
@@ -232,14 +232,10 @@ export class InfoProvider implements Disposable {
         const prev_line = editor.document.lineAt(pos.line - 1);
         const spaces = prev_line.firstNonWhitespaceCharacterIndex;
         const margin_str = [...Array(spaces).keys()].map(x => ' ').join('');
-
-        // [hack] for now, we assume that there is only ever one command per line
-        // and that the command should be inserted on the line above this one.
-
+        new_command = new_command.replace(/\n/g, '\n' + margin_str);
+        new_command = `\n${margin_str}${new_command}`;
         await editor.edit((builder) => {
-            builder.insert(
-                prev_line.range.end,
-                `\n${margin_str}${new_command}`);
+            builder.insert(prev_line.range.end, new_command);
```

Comment:
This still only allows for text to be inserted on the line above the current one, right? For Johan's use case in # I think we want to be able to insert text right at the cursor.

Reply:
ah I see

Reply:
sorry I completely missed that